### PR TITLE
add train test split

### PR DESCRIPTION
**Pull request by benisraeldan**

### CODE REVIEW

File: deepchecks/base/dataset.py [hunk 1]
```diff
@@ -381,6 +382,41 @@ def __len__(self) -> int:
...
+                         shuffle: bool = True) -> t.Tuple[TDataset, TDataset]:
+        """Split dataset into random train and test.
+
+        Args:
+            train_size (float or int):
+                If float, should be between 0.0 and 1.0 and represent the proportion
+                of the dataset to include in the test split. If int, represents the
+                absolute number of test samples. If None, the value is set to the
+                complement of the train size. If ``train_size`` is also None, it will
+                be set to 0.25.
```
Comment (ItayGabbay) [inline deepchecks/base/dataset.py:398]: Why not specifying in the method signature that 
train_size: t.Union[int, float] = 0.25?

File: deepchecks/base/dataset.py [hunk 2]
```diff
@@ -381,6 +382,41 @@ def __len__(self) -> int:
...
+            train_size (float or int):
```
Comment (ItayGabbay) [inline deepchecks/base/dataset.py:399]: test_size

File: deepchecks/base/dataset.py [hunk 3]
```diff
@@ -381,6 +382,41 @@ def __len__(self) -> int:
...
+                int, represents the absolute number of train samples. If None,
+                the value is automatically set to the complement of the test size.
+            random_state (int):
+                The random state to use for shuffling. (default=42)
+            shuffle (bool):
+                Whether or not to shuffle the data before splitting. (default=True)
+
+        Returns:
+            Train Dataset
+            Test Dataset
```
Comment (ItayGabbay) [inline deepchecks/base/dataset.py:411]: Something feels wrong in this docstring

**Review by ItayGabbay — CHANGES_REQUESTED** [review]

**Review by ItayGabbay — APPROVED** [review]

File: deepchecks/base/dataset.py [hunk 4]
```diff
@@ -381,6 +382,37 @@ def __len__(self) -> int:
...
+
+        Returns:
+            (Dataset) Dataset containing train split data.
+            (Dataset) Dataset containing test split data.
+        """
+        train_df, test_df = train_test_split(self._data,
+                                             test_size=test_size,
+                                             train_size=train_size,
+                                             random_state=random_state,
+                                             shuffle=shuffle)
```
Comment (noamzbr) [inline deepchecks/base/dataset.py:413]: add stratify

**Pull request merged**